Backend/Controllers/ProductController.cs

```diff
@@ -0,0 +1,250 @@
+﻿using Data;
+using Data.Models.ProductTables;
+using Backend.Services;
+using Lib.EntityFrameworkCore;
+using Microsoft.AspNetCore.Mvc;
+using Lib.Storage;
+using Lib.Storage.Services;
+using System.Drawing;
+using System.Drawing.Imaging;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Authorization;
+using Lib;
+
+namespace Backend.Controllers
+{
+    [Route("v1/products")]
+    [ApiController]
+    [Authorize]
+    public class ProductController : ControllerBase
+    {
+        private readonly Db db;
+
+        private readonly ImageService imageService;
+
+        private readonly IStorage storage;
+
+        private readonly Jwt jwt;
+
+
+        public ProductController(Db context, ImageService imageService, Storj storage, Jwt jwt)
+        {
+            db = context;
+            this.imageService = imageService;
+            this.storage = storage;
+            this.jwt = jwt;
+        }
+
+
+        [HttpPost]
+        public async Task<IActionResult> CreateProduct(Product newProduct)
+        {
+            var uid = User.FindFirst(Jwt.Uid)?.Value;
+            var shop = await db.Shops.QueryOne(s => s.OwnerId == uid);
+           
+            if (shop == null) return Forbid();
+
+            newProduct.IsDraft = true;
+
+            db.Products.Add(newProduct);
+
+            var saved = await db.Save();
+
+            return saved ? Ok() : Problem();
+        }
+
+    
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Product>> GetProduct(string id)
+        {
+            var product = await db.Products.QueryOne(p => p.Id == id);
+
+            var uid = User.FindFirst(Jwt.Uid)?.Value;
+            var shop = await db.Shops.QueryOne(s => s.Id == product.ShopId && s.OwnerId == uid);
+
+            if (shop == null) return Forbid();
+
+            if (product == null) return NotFound();
```

Comment:
you check if product == null, after you used product? You should do it right after getting product in line 60